src/views/ArmyDetails.vue

```diff
@@ -0,0 +1,142 @@
+<template >
+  <div>
+
+    <div class="row1">
+
+      <div class="army-details-title">
+
+      <i class="el-icon-d-arrow-left" :style={color:color} style="font-size:50px;cursor:pointer" @click="gotohome"></i>
+
+      <p class="titlle" :style="style">{{title}}</p>
+
+    </div>
+
+    </div>
+
+    <div class="row">
+     <div class="army-details"   v-for="(army, index) in armiesState" v-bind:key="index + 'armyDetailsCard'" >
+      <ArmyDetailsCard :armydata="army" :color="color"  />
+    </div>
+
+    </div>
+     <button class='add-army-btn' :style={background:color} @click="addArmy()">ADD {{title.toUpperCase()}}</button>
+  </div>
+</template>
+
+<script>
+import ArmyDetailsCard from '@/components/ArmyDetailsCard.vue'
+import { mapState } from 'vuex'
+
+export default {
+  name: 'details',
+  components: {
+    ArmyDetailsCard
+  },
+  data () {
+    return {
+      armiesState: [],
+      title: '',
+      color: '',
+      current: '',
+      type: ''
+    }
+  },
+  methods: {
+    style () {
+      return 'text-shadow:' + '0 0 20px ' + this.color
+    },
+    addArmy () {
+      this.$router.push(this.current)
+    },
+    gotohome () {
+      this.$router.push('/')
+    }
+  },
+  mounted () {
+    this.current = 'add' + this.$router.history.current.name
+    let length = this.$store.state.armyint.length
+    let armytype = this.$store.state.armyint[length - 1]
+    this.title = armytype.title
+    this.color = armytype.color
+    this.types = armytype.types
+
+    if (this.types === 1) {
+      this.$store.dispatch('getTropper')
+    } else if (this.types === 0) {
+      this.$store.dispatch('getDroid')
+    } else {
+      this.$router.push('/')
+    }
+  },
+  computed: {
+    ...mapState({
+      armies: state => {
+        return state.DroidArmy
+      },
+      armies1: state => {
+        return state.TroopersArmy
+      }
+    })
+  },
+  watch: {
```

Comment:
we internally follow the order of Vue properties/methods like.

created()
mounted()
computed()
watch: {
}
data()
methods: {
}